frontend/src/scenes/pipeline/pipelineLogic.tsx

```diff
@@ -59,11 +62,52 @@ export const pipelineLogic = kea<pipelineLogicType>([
         // it's better to add the permission checks everywhere now
         canGloballyManagePlugins: [(s) => [s.user], (user) => canGloballyManagePlugins(user?.organization)],
         canConfigurePlugins: [(s) => [s.user], (user) => canConfigurePlugins(user?.organization)],
-        canEnableNewDestinations: [
-            (s) => [s.user, s.hasAvailableFeature],
-            (user, hasAvailableFeature) =>
-                user?.is_impersonated ||
-                (canConfigurePlugins(user?.organization) && hasAvailableFeature(AvailableFeature.DATA_PIPELINES)),
+        notAllowedReasonByStageAndOperationType: [
```

Comment:
not sure this is the best way to do this, but having a consistent disabled reason / error is nice, later other things might be allowed / disallowed and that can be changed here

Reply:
I found this really hard to parse overall as a concept tbh. I don't fully get what it is trying to achieve tbh. 

My gut tells me this shouldn't be abstracted here but rather implemented in each logic for the different kinds of thing. Reason being that the benefits of abstracting it here seem very light in comparison to the high complexity of passing this object around everywhere.

For example the `NewButton` is made relatively complex as it is trying to be super abstract when in actual fact the logic to decide whether it should be shown could easily belong to the more specific logic like `pipelineDestinationsLogic` and then the `NewButton` can remain simple and not need this passed object. 

Might be worth an in person sync if this doesn't make sense 👍